### ISSUE TITLE
stderror-override and disable-shared

### ISSUE THREAD

**Issue by lhunath** [json-c/json-c#350]
First of all, `.travis.yml` suggests we should pass `--enable-stderror-override` to `./configure` but I don't think that does anything:

```
configure: WARNING: unrecognized options: --enable-strerror-override
```

Furthermore, when configuring with `--disable-shared` (I don't want a `.dylib`, only a `.a`), the test build fails:

```
  CCLD     test1
  CCLD     test2
ld: reference to symbol (which has not been assigned an address) __json_c_strerror in '_json_c_get_random_seed' from /tmp/lto.o for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [test1] Error 1
make[2]: *** Waiting for unfinished jobs....
ld: reference to symbol (which has not been assigned an address) __json_c_strerror in '_main' from /tmp/lto.o for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [test_util_file] Error 1
ld: reference to symbol (which has not been assigned an address) __json_c_strerror in '_json_c_get_random_seed' from /tmp/lto.o for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [test2] Error 1
make[1]: *** [check-am] Error 2
make: *** [check-recursive] Error 1
```

**Comment by hawicz** [comment]
The travis problem should have been fixed in commit d8fbfc7aa.

**Comment by hawicz** [comment]
The rest was fixed by commit e0e34f0a
